### ISSUE TITLE
Obscure feature requests

### ISSUE THREAD

**Issue by erkyrath** [DavidKinder/Inform6#165]
This is various pieces of information which the compiler could provide which we don't have a real use case for, but the idea came up and I want to write it down. Very low priority.

**Comment by erkyrath** [comment]
**Current routine address**

Maybe a system constant `#current_routine`. This returns `routine_start_pc` with marker `IROUTINE_MV`. (Divide by `scale_factor` in Z-code.) If used outside a routine definition, it's an error.

**Comment by erkyrath** [comment]
**Number of local variables**

Similarly, `#local_variable_count` could return `routine_locals`. Error outside a function definition.

**Comment by erkyrath** [comment]
**Absolute address of a label**

You can `@jumpabs` to a label given its address, but there's currently no way to get this. (Would only be possible in Glulx, as Z-code label addresses don't generally fit in a 16-bit value, and there is no `@jumpabs` in Z-code anyway.)

Label addresses aren't handled in backpatching; they're figured out by special code in transfer_routine(). So this is messy, and it would be messier if you want to retain the addresses of labels *outside* the code of the routine itself.

Might want a special syntax for this, maybe `..LabelName;` -- defines the label as a *global* symbol and also adds it to a list of labels to remember through compilation.

**Comment by erkyrath** [comment]
**Local variable index**

The offset where a local occurs in the locals segment. Like `#g$GLOBAL`, but for locals. The obvious spelling would be `#l$LOCAL`.

This value would be pretty hard to use. There's no way to access the locals segment except with an opcode argument, and those have fixed offsets. I guess self-modifying code could generate an opcode to do this.

**Comment by curiousdannii** [comment]
> This value would be pretty hard to use. There's no way to access the locals segment except with an opcode argument, and those have fixed offsets.

Yeah I've realised that... that might end up being the trickiest part of all this. Or maybe just the most verbose - I could do it with a long switch statement.
